v3/integrations/nrgrpc/nrgrpc_server.go

```diff
@@ -42,6 +72,167 @@ func startTransaction(ctx context.Context, app *newrelic.Application, fullMethod
+	codes.AlreadyExists:      InfoInterceptorStatusHandler,
+	codes.PermissionDenied:   WarningInterceptorStatusHandler,
+	codes.ResourceExhausted:  WarningInterceptorStatusHandler,
+	codes.FailedPrecondition: WarningInterceptorStatusHandler,
+	codes.Aborted:            WarningInterceptorStatusHandler,
+	codes.OutOfRange:         WarningInterceptorStatusHandler,
+	codes.Unimplemented:      ErrorInterceptorStatusHandler,
+	codes.Internal:           ErrorInterceptorStatusHandler,
+	codes.Unavailable:        WarningInterceptorStatusHandler,
+	codes.DataLoss:           ErrorInterceptorStatusHandler,
+	codes.Unauthenticated:    InfoInterceptorStatusHandler,
+}
+
+type handlerOption func(statusHandlerMap)
+
+//
+// `WithStatusHandler()` indicates a handler function to be used to
+// report the indicated gRPC status. Zero or more of these may be
+// given to the `Configure()`, `StreamServiceInterceptor()`, or
+// `UnaryServiceInterceptor()` functions.
+//
+// The `ErrorHandler` parameter is generally one of the provided standard
+// reporting functions:
+//  `OKInterceptorStatusHandler`      // report the operation as successful
+//  `ErrorInterceptorStatusHandler`   // report the operation as an error
+//  `WarningInterceptorStatusHandler` // report the operation as a warning
+//  `InfoInterceptorStatusHandler`    // report the operation as an informational message
+//
+// The following reporting function should only be used if you know for sure
+// you want this. It does not report the error in any way at all, but completely
+// ignores it.
+//  `IgnoreInterceptorStatusHandler`  // report the operation as successful
+//
+// Finally, if you have a custom reporting need that isn't covered by the standard
+// handler functions, you can create your own handler function as
+//   func myHandler(ctx context.Context, txn *newrelic.Transaction, s *status.Status) {
+//      ...
+//   }
+// Within the function, do whatever you need to do with the `txn` parameter to report the
+// gRPC status passed as `s`. If needed, the context is also passed to your function.
+//
+// If you wish to use your custom handler for a code such as `codes.NotFound`, you would
+// include the parameter
+//   WithStatusHandler(codes.NotFound, myHandler)
+// to your `Configure()`, `StreamServiceInterceptor()`, or `UnaryServiceInterceptor()` function.
+//
+func WithStatusHandler(c codes.Code, h ErrorHandler) handlerOption {
+	return func(m statusHandlerMap) {
+		m[c] = h
+	}
+}
+
+//
+// `Configure()` takes a list of `WithStatusHandler()` options and sets them
+// as the new default handlers for the specified gRPC status codes, in the same
+// way as if `WithStatusHandler()` were given to the `StreamServiceInterceptor()`
+// or `UnaryServiceInterceptor()` functions (q.v.); however, in this case the new handlers
+// become the default for any subsequent interceptors created by the above functions.
+//
+func Configure(options ...handlerOption) {
```

Comment:
Do you think `Configure()` might be too general a name for this? There might be other things to configure in the future.

Oh, but I suppose we could have other "With..." options that have  nothing to do with StatusHandlers, and then the Configure method name would still make sense. I'm a little fuzzy on what this would look like.

Reply:
I was pondering that too. Since it's in the nrgrpc package, it would be just configuring things within the package, but the idea was that we'd add other "With..." options for other package configuration needs, so we don't have multiple calls to make to customize the package in various ways.